hoot-core/src/main/cpp/hoot/core/algorithms/WayJoiner.cpp

```diff
@@ -185,7 +185,15 @@ void WayJoiner::_joinAtNode()
           //  Check for equivalent tags
           if (pTags == cTags || pTags.dataOnlyEqual(cTags))
           {
-            _joinWays(way, child);
+            long wid = way->getId();
+            long cid = child->getId();
+            //  Decide which ID to keep
+            if ((wid > 0 && cid < 0) ||               //  Way is the only positive ID
+                (wid < 0 && cid < 0 && wid > cid) ||  //  Larger of the two negative IDs
+                (wid > 0 && cid > 0 && wid < cid))    //  Smaller of the two positive IDs
+              _joinWays(way, child);
+            else
+              _joinWays(child, way);
```

Comment:
This is a simple way of picking which of the ways to keep and which to merge in.  Since positive IDs are from OSM, they should be preserved.  The smaller of two positive IDs is the older way in OSM and will have the most "history" so it should be kept.  The larger (or closer to zero) of two negative IDs is more likely to be an original way ID before being split in the splitting or merging processes of hoot.